src/vision/status_quo/alan_hates_writing_a_stream.md

```diff
@@ -0,0 +1,194 @@
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+[Alan][] is used to writing web server applications using async sockets, but wants to try Rust to get that signature vroom vroom.
+
+After a couple weeks learning Rust basics, Alan quickly understands `async` and `await`, and therefore has several routes built for his application that await a few things and then construct an HTTP response and send a buffered body. To build the buffered response bodies, Alan was reading a file, and then appending a signature, and putting that all into a single buffer of bytes.
+
+Eventually, Alan realizes that some responses have enormous bodies, and would like to stream them instead of buffering them fully in memory. He's *used* the `Stream` trait before. Using it was very natural, and followed a similar pattern to regular `async`/`await`:
+
+```rust
+while let Some(chunk) = body.next().await? {
+    file.write_all(&chunk).await?;
+}
+```
+
+However, _implementing_ `Stream` turns out to be rather different. While he quickly learned the simple way to turn a `File` into a `Stream` with `StreamReader`, the chaining part was much harder.
+
+### Imperatively Wrong
+
+Alan first hoped he could simply write "chain"-like stream imperatively, reusing his new knowledge of `async` and `await`, and assuming it'd be similar to JavaScript:
+
+```rust
+async* fn sign(file: ReaderStream) -> Result<Vec<u8>, Error> {
+    let mut sig = Signature::new();
+
+    while let Some(chunk) = file.next().await? {
+        sig.push(chunk.len());
+        yield Ok(chunk)
+    }
+
+    yield Ok(sig.digest().await)
+}
+```
+
+Unfortunately, that doesn't work. The compiler first complains about the `async* fn` syntax:
+
+```
+error: expected item, found keyword `async`
+  --> src/lib.rs:21:1
+   |
+21 | async* fn sign(file: ReaderStream) -> Result<Vec<u8>, Error> {
+   | ^^^^^ expected item
+```
+
+Less hopeful, Alan tries just deleting the asterisk:
+
+```
+error[E0658]: yield syntax is experimental
+  --> src/lib.rs:27:9
+   |
+27 |         yield Ok(chunk)
+   |         ^^^^^^^^^^^^^^^
+   |
+   = note: see issue #43122 <https://github.com/rust-lang/rust/issues/43122> for more information
+```
+
+After reading about how yield is experimental, and giving up reading the 100+ comments in the [linked issue](https://github.com/rust-lang/rust/issues/43122), Alan figures he's just got to implement `Stream` manually.
+
+### Implementing `Stream`
+
+Implementing a `Stream` means writing async code in a way that doesn't _feel_ like the `async fn` that Alan has written so far. He needs to write a `poll` function and it has a lot of unfamiliar concepts:
```

Comment:
In my opinion, `Stream` trait is very low level and we should not encourage most Rust developer to use it directly.  Instead, we should teach them to use combinators and other high level primitives (map, filter, etc)  to construct Stream.  

Teaching `poll_next` should be reserved for advance use cases (i.g. writing device drivers, protocol encoders)